### PR TITLE
Correct the response code of the REST Server when executing an invalid stored query

### PR DESCRIPTION
**Pull request by adamretter**
Closes https://github.com/eXist-db/exist/issues/4639

### CODE REVIEW

File: exist-core/src/main/java/org/exist/http/RESTServer.java [hunk 1]
```diff
@@ -537,9 +537,9 @@ public void doGet(final DBBroker broker, final Txn transaction, final HttpServle
                         LOG.debug(e.getMessage(), e);
                     }
                     if (MimeType.XML_TYPE.getName().equals(mimeType)) {
-                        writeXPathException(response, HttpServletResponse.SC_BAD_REQUEST, encoding, query, path, e);
+                        writeXPathException(response, HttpServletResponse.SC_INTERNAL_SERVER_ERROR, encoding, query, path, e);
```
Comment (dizzzz) [inline exist-core/src/main/java/org/exist/http/RESTServer.java:540]: is it an system error or a user error?

Comment (adamretter) [reply exist-core/src/main/java/org/exist/http/RESTServer.java:540]: @dizzzz I have only changed the codes where a user is making a request to execute a query that is already stored in the database. I think these could be classed as Internal Server Errors.

The requests that send a query to eXist-db, I have left as Bad Request error.

**Review by dizzzz — APPROVED** [review]

**Comment by adamretter** [comment]
@line-o Thanks, but... this should not have been merged yet. It is an API breaking change and has a 7.0.0 milestone against it. I am going to roll this one back and reopen the PR.

**Comment by line-o** [comment]
And why?

**Comment by line-o** [comment]
It's a fix

**Comment by adamretter** [comment]
It changes the HTTP Response codes of existing actions in a public API. Therefore it is a public API change and requires a major release.

**Comment by adamretter** [comment]
New PR is - https://github.com/eXist-db/exist/pull/4649

**Pull request merged**